datafusion/common/src/types/builtin.rs

```diff
@@ -47,3 +50,53 @@ singleton!(LOGICAL_FLOAT64, logical_float64, Float64);
 singleton!(LOGICAL_DATE, logical_date, Date);
 singleton!(LOGICAL_BINARY, logical_binary, Binary);
 singleton!(LOGICAL_STRING, logical_string, String);
+
+pub fn logical_timestamp(
```

Comment:
Some interning or similar could be beneficial here to share `Arc`s between invocations.